Sources/PointFree/Account/Account.swift

```diff
@@ -10,78 +10,115 @@ import Prelude
 import Styleguide
 import Tuple
 
-let accountResponse =
+let accountResponse: Middleware<StatusLineOpen, ResponseEnded, Tuple2<Database.User?, SubscriberState>, Data> =
   filterMap(require1 >>> pure, or: loginAndRedirect)
     <| fetchAccountData
     >-> writeStatus(.ok)
-    >-> map(lower)
-    >>> respond(
+    >-> respond(
       view: accountView,
-      layoutData: { subscription, teamInvites, teammates, emailSettings, currentUser, subscriptionStatus in
+      layoutData: { data in
         SimplePageLayoutData(
-          currentSubscriptionStatus: subscriptionStatus,
-          currentUser: currentUser,
-          data: (subscription, teamInvites, teammates, emailSettings, [], currentUser),
+          currentSubscriberState: data.subscriberState,
+          currentUser: data.currentUser,
+          data: data,
           title: "Account"
         )
     }
 )
 
-private func fetchAccountData<I, A>(
-  _ conn: Conn<I, T2<Database.User, A>>
-  ) -> IO<Conn<I, T6<Stripe.Subscription?, [Database.TeamInvite], [Database.User], [Database.EmailSetting], Database.User, A>>> {
+private func fetchAccountData<I>(
+  _ conn: Conn<I, Tuple2<Database.User, SubscriberState>>
+  ) -> IO<Conn<I, AccountData>> {
 
-  let user = get1(conn.data)
+  let (user, subscriberState) = lower(conn.data)
 
-  let subscription = user.subscriptionId
+  let userSubscription = user.subscriptionId
     .map(
-      (
-        AppEnvironment.current.database.fetchSubscriptionById
-          >>> mapExcept(requireSome)
-          >>> map(^\.stripeSubscriptionId)
-          >-> AppEnvironment.current.stripe.fetchSubscription
-        )
-        >>> ^\.run
-        >>> map(^\.right)
+      AppEnvironment.current.database.fetchSubscriptionById
+        >>> mapExcept(requireSome)
     )
-    ?? pure(nil)
+    ?? throwE(unit)
 
-  return zip4(
-    subscription.parallel,
+  let ownerSubscription = AppEnvironment.current.database.fetchSubscriptionByOwnerId(user.id)
+    .mapExcept(requireSome)
 
-    AppEnvironment.current.database.fetchTeamInvites(user.id).run.parallel
+  let owner = userSubscription
+    .flatMap { subscription in
+      subscription.userId == user.id
+        ? pure(user)
+        : AppEnvironment.current.database.fetchUserById(subscription.userId)
+    }
+    .mapExcept(requireSome)
+
+  let subscription = userSubscription <|> ownerSubscription
+
+  let stripeSubscription = subscription
+    .map(^\.stripeSubscriptionId)
+    .flatMap(AppEnvironment.current.stripe.fetchSubscription)
+
+  let everything = zip7(
```

Comment:
looks like swift 4.1 fixes the linux bug that caused crashes when we `zip5`'d stuff. now we can even `zip7`!